lib/rom/mapper.rb

```diff
@@ -56,7 +56,16 @@ def self.register_processor(processor)
     #
     # @api private
     def self.build(header = self.header, processor = :transproc)
-      new(Mapper.processors.fetch(processor).build(header), header)
+      if steps.empty?
+        new(Mapper.processors.fetch(processor).build(header), header)
+      else
+
+        step_transformers = steps.map do |s|
+          new(Mapper.processors.fetch(processor).build(s.header), s.header).transformer
+        end.inject(:+)
```

Comment:
This assumes transformers respond to `:+` I wonder if we could avoid that somehow